docs/fhir-guide/FHIR 101 - Practical Guide.ipynb

```diff
@@ -0,0 +1,2966 @@
+    "    \"text\": {\n",
+    "        \"status\": \"generated\",\n",
+    "        \"div\": \"<div xmlns=\\\"http://www.w3.org/1999/xhtml\\\"><p>A valid patient resource for testing purposes</p></div>\"\n",
+    "    },\n",
+    "    \"meta\": {\n",
+    "        \"profile\": [\"http://fhir.kids-first.io/StructureDefinition/Patient\"]\n",
+    "    },\n",
+    "    \"gender\": \"male\"\n",
+    "}"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "heading_collapsed": true,
+    "hidden": true
+   },
+   "source": [
+    "#### Validate the Patient Resource 🖐"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "hidden": true
+   },
+   "source": [
+    "To test the `Patient` resource against the Kids First Patient model you must declare that this resource conforms to the `Patient` `StructureDefinition` like this:\n",
+    "\n",
+    "```json\n",
+    "\"meta\": {\n",
+    "    \"profile\": [\"http://fhir.kids-first.io/StructureDefinition/Patient\"]\n",
+    "}\n",
+    "```"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "metadata": {
+    "code_folding": [
+     0
+    ],
+    "hidden": true
+   },
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "{'request_url': 'http://hapi.fhir.org/baseR4/Patient/$validate',\n",
+      " 'response': {'issue': [{'code': 'informational',\n",
+      "                         'diagnostics': 'No issues detected during validation',\n",
+      "                         'severity': 'information'}],\n",
+      "              'resourceType': 'OperationOutcome',\n",
+      "              'text': {'div': '<div '\n",
+      "                              'xmlns=\"http://www.w3.org/1999/xhtml\"><h1>Operation '\n",
+      "                              'Outcome</h1><table border=\"0\"><tr><td '\n",
+      "                              'style=\"font-weight: '\n",
+      "                              'bold;\">INFORMATION</td><td>[]</td><td><pre>No '\n",
+      "                              'issues detected during validation</pre></td>\\n'\n",
+      "                              '\\t\\t\\t\\t\\t\\n'\n",
+      "                              '\\t\\t\\t\\t\\n'\n",
+      "                              '\\t\\t\\t</tr>\\n'\n",
+      "                              '\\t\\t</table>\\n'\n",
+      "                              '\\t</div>',\n",
+      "                       'status': 'generated'}},\n",
+      " 'status_code': 200}\n",
+      "Validate http://hapi.fhir.org/baseR4/Patient/$validate, kidsfirst-test-patient\n",
+      "PUT http://hapi.fhir.org/baseR4/Patient/kidsfirst-test-patient\n"
+     ]
+    }
+   ],
+   "source": [
+    "def validate_rs(payload, expected_success=True, create_on_success=True, print_result=False):\n",
+    "    resource_type = payload.get('resourceType')\n",
+    "    \n",
+    "    # Validate by POSTing to the /Patient/$validate endpoint\n",
```

Comment:
But why?
If it's invalid, will creating the resource not fail with a validation error?

Reply:
I think it depends on the server, but if validation is turned off then POSTing/creating invalid resources will succeed. I think the `$validate` endpoint will always validate regardless of whether validation is turned on or off. I THINK...

Reply:
Can you mention that?

Reply:
I mentioned it in the Model Validation section before the tutorial